app/models/project.rb

```diff
@@ -225,6 +225,7 @@ def notifications_count(user)
   def generate_report_pdf(user, team, pdf_name, obj_filenames = nil)
     ActionController::Renderer::RACK_KEY_TRANSLATION['warden'] ||= 'warden'
     proxy = Warden::Proxy.new({}, Warden::Manager.new({}))
+    proxy.set_user(user, scope: :user, store: false)
     renderer = ApplicationController.renderer.new(warden: proxy)
 
     report = Report.generate_whole_project_report(self, user, team)
```

Comment:
I would add a `begin`-`rescue` block up until the `report.destroy` line, and put the `report.destroy` in the `rescue` block, just to make sure the `report` is surely deleted in case we encounter another bug in the future.